internal/controller/linstorcluster_controller.go

```diff
@@ -210,6 +211,22 @@ func (r *LinstorClusterReconciler) reconcileAppliedResource(ctx context.Context,
 		}
 	}
 
+	// Extract information which config applied to which satellites
+	var currentSatellites piraeusiov1.LinstorSatelliteList
+	err = r.Client.List(ctx, &currentSatellites, &client.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	configAppliedToSatellites := make(map[string][]string)
+	for i := range currentSatellites.Items {
+		configs := strings.Split(currentSatellites.Items[i].Annotations[vars.AppliedConfigurationAnnotation], ",")
```

Comment:
It seems strange to me that we store and then retrieve this data to populate this field. Can't we pass the information about which configs applied to which satellites back from `kustomizeLinstorSatellite`?

Reply:
Done. I did not do it originally, as changing the return value changes a lot of error-handling code, but that is a bad reason to pass this information through some obscure other means. :sweat_smile: